modules/packages/LinearAlgebra.chpl

```diff
@@ -639,7 +639,39 @@ proc inner(A: [?Adom], B: [?Bdom]) {
   if Adom.size != Bdom.size then
     halt("Mismatched size in inner multiplication");
 
-  return + reduce(A[..]*B[..]);
+  var results: [0..#numLocales] etype = 0;
```

Comment:
I think `Locales.domain` is called `LocaleSpace`.

I am concerned about this computation going over all locales because this will hurt the performance when `A` and `B` are not distributed. At the very least, this computation needs to stay local in the known cases such as --local compilation and when `A` and `B` are default rectangular. For known distributions, such as Block and Cyclic, the computation should use `targetLocales` instead of `Locales`. While in the common case the two are the same, I am uneasy about introducing a hidden performance bug noticeable in some runs over lots of locales.

I believe that the computation written as `+ reduce(A*B)` distributes/parallelizes using `A`'s leader, i.e. over `A`'s locales. BTW.

Reply:
@vasslitvinov - Thanks for pointing out `LocaleSpace`, I never remember that :(

Also, good point on using `targetLocales`.. `Locales` should change to `A.targetLocales()` : https://chapel-lang.org/docs/technotes/subquery.html?highlight=targetlocales#targetlocales

I agree that there probably needs to be a check to whether they're default rectangular (non-distributed or not) I was hoping that we would get there once we run some performance tests and confirmed that that's the case

@wongalvis14 - You updated your PR message with 1 locale. However, it is still different than compiling it with `--local` flag or compiling with a chapel that's built with `CHPL_COMM=none`. Is your 1-locale result above comes from any of these? Or did you simply do `-nl 1` while running the benchmark?

I'd like to do some performance tests myself before this PR is merged.